### PR TITLE
networkfilter.py: remember previous IP addresses of user

### PR DESCRIPTION
**Pull request by mathiascode**
- If an IP-banned user changes their IP address, keep the filter for the old IP address in place until the user is unbanned.
- Various code cleanups

### CODE REVIEW

**Comment by mathiascode** [comment]
I'm using just the username as the placeholder instead of `?.?.?.?`, seemed a bit cleaner to handle. It doesn't look ideal in the IP address column, but perhaps usernames could be replaced with "Unknown" before appending to the list view somehow.

**Comment by slook** [comment]
> perhaps usernames could be replaced with "Unknown" before appending to the list view somehow

The idiom `user in ip_addresses` is misleading to read in the code, as well as looking a bit dumb in the list. Suggest adding something like `f"{user} (Unknown)"` instead of just `user`.

> Let me know if I broke something that's needed for CLI.

Nothing seems to be broken, only very minor changes are required to my draft branch.

**Comment by mathiascode** [comment]
> The idiom `if user and user in ip_addresses` is misleading to read in the code, as well as looking a bit dumb in the list. Suggest adding something like `f"{user} (Unknown)"` instead of just `user`, would be a good idea, yes.

How about `username in ip_addresses` and a comment next to it?

I was mainly thinking about replacing the whole username with `Unknown` in the preferences GUI. If we just append `(Unknown)`, we will have problems with translations anyway.

File: pynicotine/core.py [hunk 1]
```diff
@@ -333,6 +333,7 @@ def _server_disconnect(self, manual_disconnect=False):
         self.user_status = slskmessages.UserStatus.OFFLINE
 
         # Clean up connections
+        self.user_addresses.clear()
```
Comment (slook) [inline pynicotine/core.py:336]: Crash on quit, after interacting with a private chat of a non-existant user who is never online:

```
Error in sys.excepthook:
Traceback (most recent call last):
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/mainwindow.py", line 362, in on_window_active_changed
    self.privatechat.clear_notifications()
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/privatechat.py", line 133, in clear_notifications
    self.window.application.notifications.clear("private", user=user)
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/widgets/notifications.py", line 71, in clear
    if core.notifications.remove_hilite_item(location, item):
  File "/home/user/Git/slook/nicotine-plus/pynicotine/notifications.py", line 57, in remove_hilite_item
    if item not in self.chat_hilites[location]:
KeyError: 'private'

Original exception was:
Traceback (most recent call last):
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/mainwindow.py", line 362, in on_window_active_changed
    self.privatechat.clear_notifications()
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/privatechat.py", line 133, in clear_notifications
    self.window.application.notifications.clear("private", user=user)
  File "/home/user/Git/slook/nicotine-plus/pynicotine/gtkgui/widgets/notifications.py", line 71, in clear
    if core.notifications.remove_hilite_item(location, item):
  File "/home/user/Git/slook/nicotine-plus/pynicotine/notifications.py", line 57, in remove_hilite_item
    if item not in self.chat_hilites[location]:
KeyError: 'private'
/usr/lib/python3/dist-packages/gi/overrides/Gio.py:43: Warning: Source ID 9479 was not found when attempting to remove it
  return Gio.Application.run(self, *args, **kwargs)
```

**Comment by mathiascode** [comment]
Or maybe something like this, if we don't want to mess with the listview append code and also want to save space:

Screenshot from 2023-01-26 02-40-25

File: pynicotine/core.py (hunk 1 above)
Comment (mathiascode) [reply pynicotine/core.py:336]: That's a different issue.

**Comment by slook** [comment]
Yes that fits better

File: pynicotine/core.py (hunk 1 above)
Comment (slook) [reply pynicotine/core.py:336]: Moved to discussion https://github.com/nicotine-plus/nicotine-plus/discussions/2145#discussioncomment-4783141

**Comment by mathiascode** [comment]
> What happens when you hit Remove on a user who has multiple address entries, does it just remove the one item or all of the matching ones?

One item

**Comment by mathiascode** [comment]
Yep. In the future, we could sort the listviews by username to make it easier to find IP addresses for the same username.

**Pull request merged**